modules/channel-teams/src/renderers/choices.ts

```diff
@@ -34,7 +34,7 @@ export class TeamsChoicesRenderer implements ChannelRenderer<TeamsContext> {
             return {
               title: reply.title as string,
               type: 'messageBack',
-              value: reply.value,
+              value: { value: reply.value, renderer: 'choices' },
```

Comment:
A renderer is used to convert a botpress payload to a channel specific payload at the time it is sent out to the channel. So you can't set a property here and expect to have it set when you receive a message. Renderers don't affect incoming payloads

Reply:
The `value` field we send to Teams is returned to Botpress as-is (see https://docs.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-actions#messageback)